pkg/apis/servicecatalog/plan_reference.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicecatalog
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ClassSpecified checks that at least one class field is set.
+func (pr PlanReference) ClassSpecified() bool {
+	return pr.ClusterServiceClassExternalName != "" ||
+		pr.ClusterServiceClassExternalID != "" ||
+		pr.ClusterServiceClassName != ""
+}
+
+// PlanSpecified checks that at least one plan field is set.
+func (pr PlanReference) PlanSpecified() bool {
+	return pr.ClusterServicePlanExternalName != "" ||
+		pr.ClusterServicePlanExternalID != "" ||
+		pr.ClusterServicePlanName != ""
+}
+
+// GetSpecifiedClass returns the user-specified class value from either:
+// * ClusterServiceClassExternalName
+// * ClusterServiceClassExternalID
+// * ClusterServiceClassName
+func (pr PlanReference) GetSpecifiedClass() string {
```

Comment:
Is this for formatting purposes? I get the impression that it is, and that's fine. If it is, we should specify in the method godoc that this is for presentation purposes only, since we are conflating different types of information in the return value of this method.

Reply:
Yes it's for formatting in our logs and in the svcat cli. I'll add that to the godoc.